UpdatedImageProcessing/TargetDetection/mass_target_detector.py

```diff
@@ -0,0 +1,136 @@
+    AutomaticTester.save_result(true_positive_list, false_positive_list, index_number)
+
+class MassTargetDetector(object):
+
+    #multiprocessing pool
+    @staticmethod
+    def detect_mass_target():
+        target_map_answer_index_combo_list = []
+
+        for index in range(1, TargetDetectionSettings.NUMBER_OF_TARGET_MAPS + 1):
+            current_target_map = TargetDetectionSettings.TARGET_MAPS_PATH + "/" + str(index) + ".jpg"
+            current_target_map_answer = TargetDetectionSettings.TARGET_MAPS_ANSWERS_PATH + "/" + str(index) + ".json"
+            target_map_answer_index_combo_list.append([current_target_map, current_target_map_answer, index])
+
+        pool = multiprocessing.Pool(processes = multiprocessing.cpu_count())
+        maps_per_process = TargetDetectionSettings.NUMBER_OF_TARGET_MAPS / multiprocessing.cpu_count()
+
+    @staticmethod
+    def run_mass_target_detector(maps_per_process, process_number):
+        for index in range(1, maps_per_process + 1):
+            current_target_map = TargetDetectionSettings.TARGET_MAPS_PATH + "/" + str(index + process_number) + ".jpg"
+            current_target_map_answers = TargetDetectionSettings.TARGET_MAPS_ANSWERS_PATH + "/" + str(index + process_number) + ".json"
+
+            positive_list = BlobDetector(current_target_map).detect_blobs()
+
+            if (len(positive_list) > 15):
+                positive_list = FalsePositiveEliminator.eliminate_overrepeated_colors(current_target_map, positive_list)
+
+            positive_list = FalsePositiveEliminator.eliminate_overlapping_blobs(positive_list)
+            positive_list = FalsePositiveEliminator.eliminate_by_surrounding_color(current_target_map, positive_list)
+            print positive_list
+            #return AutomaticTester(positive_list, current_target_map_answers).run_automatic_tester()
+
+    #standard multiprocessing
+    @staticmethod
+    def detect_mass_target():
+        cpu_count = multiprocessing.cpu_count()
+        maps_per_process = (TargetDetectionSettings.NUMBER_OF_TARGET_MAPS / cpu_count)
+        start_time = timeit.default_timer()
+
+        jobs = []
+        for index in range(cpu_count):
+            starting_index = index * int(maps_per_process)
+            target_detection_process = multiprocessing.Process(target=MassTargetDetector.run_mass_target_detector, args=(maps_per_process, starting_index))
+            jobs.append(target_detection_process)
+            target_detection_process.start()
+
+        for job in jobs:
+            job.join()
+
+        TargetDetectionLogger.log("Target Detection Results saved at: " + TargetDetectionSettings.TARGET_DETECTION_SAVE_PATH)
+
+        print("====================================")
+        print("Total number of target_maps detected:", len(os.listdir(TargetDetectionSettings.TARGET_DETECTION_SAVE_PATH)))
+        print("Total elapsed time (sec):", timeit.default_timer() - start_time)
+        print("====================================")
+"""
```

Comment:
We should test this with multiprocessing enabled. That way we wouldn't have to load the images sequentially, which might actually be a significant improvement. However, it does create a RAM usage issue if we are trying to work with a ton of images...

Reply:
I cannot run the code with multiprocessing. We have discussed the errors during the break.